### PR TITLE
Changeset pr manual merge

### PR DESCRIPTION
**Pull request by a8trejo**
## Description
Changeset pr merge flag

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the patterns of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Reviewers

----

> [!IMPORTANT]
> Remove `--auto` flag from `gh pr merge` and update changelog format in GitHub Actions workflows.
> 
>   - **Workflow Changes**:
>     - Remove `--auto` flag from `gh pr merge` in `changeset-release.yml`.
>     - Update `run-name` in `changeset-release.yml` to "Ready for Review".
>   - **Changelog**:
>     - Adjust formatting for version 2.2.17 in `CHANGELOG.md`.
>   - **Changeset Config**:
>     - Add `getReleaseSummary` function in `changelog-config.js`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooVetGit%2FRoo-Cline&utm_source=github&utm_medium=referral)<sup> for c9e07cca58b7e6db9c5c4d0208b263c529c4d82a. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: .github/workflows/changeset-release.yml [hunk 1]
```diff
@@ -69,23 +69,22 @@ jobs:
...
         with:
           review-message: "I'm approving since it's a bump version PR"
-
-      # Enable auto-merge for the PR
-      - name: Enable automerge on PR
-        run: gh pr merge --merge --auto ${{ github.event.pull_request.number }}
+      
+      # Auto-merge PR
+      - name: Automerge on PR
+        if: false # Needs enablePullRequestAutoMerge in repo settings to work
```
Comment (a8trejo) [inline .github/workflows/changeset-release.yml:87]: Skipping auto-merge step

File: .github/workflows/marketplace-publish.yml [hunk 2]
```diff
@@ -1,16 +1,26 @@
 name: Publish Extension
 on:
-  push:
-    branches: ["main"]
+  pull_request:
```
Comment (a8trejo) [inline .github/workflows/marketplace-publish.yml:3]: Changing the event to have access to the merged PR github context

File: .github/workflows/marketplace-publish.yml [hunk 3]
```diff
@@ -1,16 +1,26 @@
...
+    types: [closed]
   workflow_dispatch:
 
+env:
+  GIT_REF: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || 'main' }}
```
Comment (a8trejo) [inline .github/workflows/marketplace-publish.yml:8]: defaulting to `main` so no feature branches can be manually published

File: .github/workflows/marketplace-publish.yml [hunk 4]
```diff
@@ -1,16 +1,26 @@
...
+
 jobs:
   publish-extension:
     runs-on: ubuntu-latest
-    if: ${{ github.actor == 'R00-B0T' || github.event_name == 'workflow_dispatch' }}
+    if: >
+        ( github.event_name == 'pull_request' &&
+        github.event.pull_request.base.ref == 'main' &&
+        contains(github.event.pull_request.title, 'Changeset version bump') ) ||
+        github.event_name == 'workflow_dispatch'
```
Comment (a8trejo) [inline .github/workflows/marketplace-publish.yml:17]: Still allowing manual runs of this workflow OR only changeset PRs as trigger

**Review by mrubens — APPROVED** [review]
Looks good to me but would appreciate eyes from @ColemanRoo as well

**Review by ColemanRoo — APPROVED** [review]
Looks good to me too!

**Pull request merged**